examples/cpp/cmake/Findgeode-native.cmake

```diff
@@ -0,0 +1,112 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#.rst:
+# Findgeode-native
+# ---------
+#
+# Find the Geode Native headers and library.
+#
+# Imported Targets
+# ^^^^^^^^^^^^^^^^
+#
+# This module defines the following :prop_tgt:`IMPORTED` targets:
+#
+# ``apache.geode::cpp``
+#
+# Result Variables
+# ^^^^^^^^^^^^^^^^
+#
+# This module will set the following variables in your project:
+#
+# ``GEODE_NATIVE_INCLUDE_DIRS``
+#   where to find CacheFactory.hpp, etc.
+# ``GEODE_NATIVE_LIBRARIES``
+#   the libraries to link against to use Geode Native.
+# ``GEODE_NATIVE_FOUND``
+#   true if the Geode Native headers and libraries were found.
+#
+
+
+set(_GEODE_NATIVE_ROOT "")
+if(GEODE_NATIVE_ROOT AND IS_DIRECTORY "${GEODE_NATIVE_ROOT}")
+    set(_GEODE_NATIVE_ROOT "${GEODE_NATIVE_ROOT}")
+    set(_GEODE_NATIVE_ROOT_EXPLICIT 1)
+else()
+    set(_ENV_GEODE_NATIVE_ROOT "")
+    if(DEFINED ENV{GFCPP})
+        file(TO_CMAKE_PATH "$ENV{GFCPP}" _ENV_GEODE_NATIVE_ROOT)
+    endif()
+    if(_ENV_GEODE_NATIVE_ROOT AND IS_DIRECTORY "${_ENV_GEODE_NATIVE_ROOT}")
+        set(_GEODE_NATIVE_ROOT "${_ENV_GEODE_NATIVE_ROOT}")
+        set(_GEODE_NATIVE_ROOT_EXPLICIT 0)
+    endif()
+    unset(_ENV_GEODE_NATIVE_ROOT)
+endif()
+
+set(_GEODE_NATIVE_HINTS)
+set(_GEODE_NATIVE_PATHS)
+
+if(_GEODE_NATIVE_ROOT)
+    set(_GEODE_NATIVE_HINTS ${_GEODE_NATIVE_ROOT})
+else()
+    set(_GEODE_NATIVE_PATHS (
+        "/opt/local"
+        "/usr/local"
+        "../../../../"
```

Comment:
WTH is this path?

Reply:
the ../../../../ path is the relative path. Currently, the examples are part of the native client install, it only makes sense to do a little check relative to itself. Once the examples are fully separated, this check will become unnecessary.